uvicorn/protocols/http/httptools_impl.py

```diff
@@ -391,25 +418,27 @@ async def run_asgi(self, app):
                 self.logger.error(msg)
                 self.transport.close()
         finally:
-            self.on_response = None
-
-    async def send_500_response(self):
-        await self.send(
-            {
-                "type": "http.response.start",
-                "status": 500,
-                "headers": [
-                    (b"content-type", b"text/plain; charset=utf-8"),
-                    (b"connection", b"close"),
-                ],
-            }
-        )
-        await self.send(
-            {"type": "http.response.body", "body": b"Internal Server Error"}
-        )
+            self.on_response = lambda: None
```

Comment:
This went from `self.on_response = None` to self.on_response = lambda: None`. The new version sounds more correct. Was this a bug?

Reply:
Good question. That was introduced on https://github.com/encode/uvicorn/pull/189. I don't know if the line itself is really necessary (?)...